src/main/java/com/onarandombox/MultiverseCore/commands/ListCommand.java

```diff
@@ -0,0 +1,126 @@
+import com.onarandombox.MultiverseCore.commandtools.flags.CommandValueFlag;
+import com.onarandombox.MultiverseCore.commandtools.flags.ParsedCommandFlags;
+import com.onarandombox.MultiverseCore.display.ContentDisplay;
+import com.onarandombox.MultiverseCore.display.filters.ContentFilter;
+import com.onarandombox.MultiverseCore.display.filters.DefaultContentFilter;
+import com.onarandombox.MultiverseCore.display.filters.RegexContentFilter;
+import com.onarandombox.MultiverseCore.display.handlers.PagedSendHandler;
+import com.onarandombox.MultiverseCore.display.parsers.ListContentProvider;
+import org.bukkit.ChatColor;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+@CommandAlias("mv")
+public class ListCommand extends MultiverseCoreCommand {
+    public ListCommand(@NotNull MultiverseCore plugin) {
+        super(plugin);
+
+        registerFlagGroup(CommandFlagGroup.builder("mvlist")
+                .add(CommandValueFlag.builder("--filter", ContentFilter.class)
+                        .addAlias("-f")
+                        .context((value) -> {
+                            try {
+                                return RegexContentFilter.fromString(value);
+                            } catch (IllegalArgumentException e) {
+                                throw new InvalidCommandArgument("Invalid filter: " + value);
+                            }
+                        })
+                        .build())
+                .add(CommandValueFlag.builder("--page", Integer.class)
+                        .addAlias("-p")
+                        .context((value) -> {
+                            try {
+                                return Integer.parseInt(value);
+                            } catch (NumberFormatException e) {
+                                throw new InvalidCommandArgument("Invalid page number: " + value);
+                            }
+                        })
+                        .build())
+                .build());
+    }
+
+    @Subcommand("list")
+    @CommandPermission("multiverse.core.list.worlds")
+    @CommandCompletion("@flags:groupName=mvlist")
+    @Syntax("--filter [filter] --page [page]")
+    @Description("Displays a listing of all worlds that you can enter.")
+    public void onListCommand(BukkitCommandIssuer issuer,
+
+                              @Syntax("--filter [filter] --page [page]")
+                              @Description("Filters the list of worlds by the given regex and displays the given page.")
+                              String[] flags
+    ) {
+        ParsedCommandFlags parsedFlags = parseFlags(flags);
+        ContentDisplay.create()
+                .addContent(ListContentProvider.forContent(getListContents(issuer)))
+                .withSendHandler(PagedSendHandler.create()
+                        .withHeader("%s====[ Multiverse World List ]====", ChatColor.GOLD)
```

Comment:
You can do some trickery with strikethrough chat formatting to get single lines without breaks, might be worth trying to do that.

Reply:
hm but how do you center align it tho

Reply:
```html
<gold><strikethrough>        </strikethrough> [ Multiverse World List ] <strikethrough>        </strikethrough>
```
There is what it would look like in MiniMessage. [Here is the viewer link](https://webui.adventure.kyori.net/?mode=chat_closed&input=%3Cgold%3E%3Cstrikethrough%3E%20%20%20%20%20%20%20%20%3C%2Fstrikethrough%3E%20%5B%20Multiverse%20World%20List%20%5D%20%3Cstrikethrough%3E%20%20%20%20%20%20%20%20%3C%2Fstrikethrough%3E&bg=grass), it displays the strike-through lines thinner than in game though, so keep that in mind.